lib/diplomat/cursor.ex

```diff
@@ -0,0 +1,17 @@
+defmodule Diplomat.Cursor do
+  @type t :: %__MODULE__{value: String.t()}
+
+  defstruct [:value]
+
+  @spec new(String.t(), Keyword.t()) :: t
+  def new(cursor_value, opts \\ [])
+  def new(cursor_value, encode: true), do: %__MODULE__{value: encode(cursor_value)}
```

Comment:
I'm not 100% sold on the idea of using a flag `encode` to signal that we want to encode the cursor value.

Realistically we have the following scenarios:

* We receive a cursor from DataStore which we want to encode and return to the user
* We receive an encoded cursor from the user which we want to pass to DataStore

This flag allows us to cover both of them. If anyone has a different approach to this I'd be interested in hearing it 😊